src/components/TokenTranslationUpdate/TokenTranslationUpdate.js

```diff
@@ -0,0 +1,275 @@
+import TextField from '@material-ui/core/TextField';
+import Grid from '@material-ui/core/Grid';
+import React, { useState, useEffect } from 'react';
+import NavigateNextRoundedIcon from '@material-ui/icons/NavigateNext';
+import NavigateBeforeIcon from '@material-ui/icons/NavigateBefore';
+import DoneIcon from '@material-ui/icons/Done';
+import DoneAllIcon from '@material-ui/icons/DoneAll';
+import { makeStyles } from '@material-ui/core/styles';
+import { API } from '../../store/api';
+import SnackBar from '../SnackBar/SnackBar.js';
+import Tooltip from '@material-ui/core/Tooltip';
+import Dialog from '@material-ui/core/Dialog';
+import { Button } from '@material-ui/core';
+import IconButton from '@material-ui/core/IconButton';
+import CloseIcon from '@material-ui/icons/Close';
+
+const useStyles = makeStyles({
+  icon: {
+    margin: 0,
+  },
+
+  tokenReplace: {
+    textAlign: 'center',
+  },
+  tokeninfo: {
+    marginTop: '10px',
+  },
+  closeButton: {
+    padding:0,
+  },
+  closeButtonGrid:{
+    textAlign:'right',
+    margin:'4px'
+  },
+  DialogContainer:{
+    width:'320px',
+    paddingBottom:'15px'
+  },
+  replaceButton:{
+    marginTop:'3px',
+    paddingTop:'2px',
+    paddingRight:'8px',
+    paddingLeft:'8px',
+    paddingBottom:'2px'
+  }
+});
+
+
+export default function TokenTranslationUpdate(props) {
+  const classes = useStyles(props);
+  const projectId = props.projectId;
+  const token = props.tokenDetail.token;
+  const [tokenTranslation, setTokenTranslation] = useState('');
+  const [occurance, setOccurance] = useState('');
+  const [tokenSelected, setTokenSelected] = useState(1);
+  const tokenTranslationCount = props.tokenDetail.occurrences.length;
+  const [responseStatus, setResponseStatus] = React.useState([]);
+  const [open, setOpen] = React.useState(false);
+
+  const handleClose = () => {
+    setResponseStatus([false]);
+  };
+
+  const handleClickOpen = () => {
+    setOpen(true);
```

Comment:
we can check if any tokens are translated before opening the dialog, if no tokens are translated then we can call replaceAll directly without asking user